dicemix.go

```diff
@@ -351,7 +351,11 @@ func (s *Session) run(ctx context.Context, n int, br *messages.BR) error {
 	}
 
 	// Derive shared secret keys
-	r.srKP, r.dcKP = dcnet.SharedKeys(s.kx, ecdh, s.sid, MessageSize, r.run, s.myStart, s.mcount)
+	r.srKP, r.dcKP, err = dcnet.SharedKeys(s.kx, ecdh, s.sid, MessageSize, r.run, s.myStart, s.mcount)
+	if err != nil {
+		s.log.Print(err)
+		return s.clientRunFail(ctx, rs)
```

Comment:
this is the correct thing to return here, but we need to update the messaging first.  The server is receiving a SR message here, but the SR message does not have any RevealSecrets field, and that will need to be added, and then checked by the server (similar to how it checks this field in the DC message).